### PR TITLE
Fix issue with incorrect behaviour in AlertManeuver

### PR DESCRIPTION
**Pull request by okozlovlux**
- fixed issue when in AlertManeuver SDL responds GENERIC_ERROR instead of
    INVALID_DATA when soft button has Type is Image or Both and Text is
    whitespace or \t or \n or empty
  - fixed failed UTs

Related to [Issue-980](https://github.com/smartdevicelink/sdl_core/issues/980)

### CODE REVIEW

**Comment by okozlovlux** [comment]
@LuxoftAKutsan @pold500 @indlu @istoilovgithub  please review

File: src/components/application_manager/src/commands/mobile/alert_maneuver_request.cc [hunk 1]
```diff
@@ -238,6 +238,25 @@ bool AlertManeuverRequest::IsWhiteSpaceExist() {
       }
     }
   }
+  if ((*message_)[strings::msg_params].keyExists(strings::soft_buttons)) {
+    const smart_objects::SmartArray* sb_array =
```
Comment (LuxoftAKutsan) [inline src/components/application_manager/src/commands/mobile/alert_maneuver_request.cc]: @okozlovlux check this pointer with DCHECK (soft buttons should be array, if it is not - our API parser works not correctly)

File: src/components/application_manager/src/commands/mobile/alert_maneuver_request.cc [hunk 2]
```diff
@@ -238,6 +238,25 @@ bool AlertManeuverRequest::IsWhiteSpaceExist() {
...
+        (*message_)[strings::msg_params][strings::soft_buttons].asArray();
+
+    smart_objects::SmartArray::const_iterator it_sb = sb_array->begin();
+    smart_objects::SmartArray::const_iterator it_sb_end = sb_array->end();
+
+    for (; it_sb != it_sb_end; ++it_sb) {
+      str = (*it_sb)[strings::text].asCharArray();
```
Comment (LuxoftAKutsan) [inline src/components/application_manager/src/commands/mobile/alert_maneuver_request.cc]: @okozlovlux as I understand yo need to get str only for checking, please use lose local  variable, always try to minimize scope of variables usage.

File: src/components/application_manager/src/commands/mobile/alert_maneuver_request.cc [hunk 3]
```diff
@@ -238,6 +238,25 @@ bool AlertManeuverRequest::IsWhiteSpaceExist() {
...
+      if (strlen(str) && !CheckSyntax(str)) {
+        LOG4CXX_ERROR(logger_, "Invalid soft_buttons syntax check failed");
+        return true;
+      } else if (!CheckSyntax(str, false)) {
+        LOG4CXX_ERROR(logger_, "Invalid soft_buttons syntax check failed");
```
Comment (LuxoftAKutsan) [inline src/components/application_manager/src/commands/mobile/alert_maneuver_request.cc]: @okozlovlux Confusd logic , please make it more clear, at lease pass 'false' as variable with name. And log not the same errors,  and split it ti 3 if's with returns

File: src/components/application_manager/src/commands/mobile/alert_maneuver_request.cc (hunk 1 above)
Comment (okozlovlux) [inline src/components/application_manager/src/commands/mobile/alert_maneuver_request.cc]: @LuxoftAKutsan , fix in c4915fc

File: src/components/application_manager/src/commands/mobile/alert_maneuver_request.cc (hunk 2 above)
Comment (okozlovlux) [inline src/components/application_manager/src/commands/mobile/alert_maneuver_request.cc]: @LuxoftAKutsan , fix in c4915fc

File: src/components/application_manager/src/commands/mobile/alert_maneuver_request.cc (hunk 3 above)
Comment (okozlovlux) [inline src/components/application_manager/src/commands/mobile/alert_maneuver_request.cc]: @LuxoftAKutsan , fix in c4915fc

File: src/components/application_manager/src/commands/mobile/alert_maneuver_request.cc [hunk 4]
```diff
@@ -239,18 +239,22 @@ bool AlertManeuverRequest::IsWhiteSpaceExist() {
...
 
     for (; it_sb != it_sb_end; ++it_sb) {
+      const char* str = NULL;
       str = (*it_sb)[strings::text].asCharArray();
-      if (strlen(str) && !CheckSyntax(str)) {
-        LOG4CXX_ERROR(logger_, "Invalid soft_buttons syntax check failed");
-        return true;
-      } else if (!CheckSyntax(str, false)) {
+      const bool allow_empty_line = false;
+      if ((strlen(str) && !CheckSyntax(str)) ||
```
Comment (LuxoftAKutsan) [inline src/components/application_manager/src/commands/mobile/alert_maneuver_request.cc]: @okozlovlux make conditions simpler 
```
const bool allow_empty_line =  (strlen(str) == 0);
if (!(CheckSyntax(str, allow_empty_line ))  {
// do staff 
}
```

File: src/components/application_manager/src/commands/mobile/alert_maneuver_request.cc [hunk 5]
```diff
@@ -239,18 +239,22 @@ bool AlertManeuverRequest::IsWhiteSpaceExist() {
...
+            smart_objects::SmartType_Array,
+        true);
     const smart_objects::SmartArray* sb_array =
         (*message_)[strings::msg_params][strings::soft_buttons].asArray();
 
     smart_objects::SmartArray::const_iterator it_sb = sb_array->begin();
     smart_objects::SmartArray::const_iterator it_sb_end = sb_array->end();
 
     for (; it_sb != it_sb_end; ++it_sb) {
+      const char* str = NULL;
```
Comment (okozlovlux) [inline src/components/application_manager/src/commands/mobile/alert_maneuver_request.cc]: @LuxoftAKutsan , fixed in 	33b7da2

File: src/components/application_manager/src/commands/mobile/alert_maneuver_request.cc (hunk 4 above)
Comment (okozlovlux) [inline src/components/application_manager/src/commands/mobile/alert_maneuver_request.cc]: @LuxoftAKutsan , checked again. Fixed in 	33b7da2

Comment (okozlovlux) [inline src/components/application_manager/src/commands/mobile/alert_maneuver_request.cc]: @LuxoftAKutsan , checked again. Fixed in 	33b7da2

Comment (LuxoftAKutsan) [inline src/components/application_manager/src/commands/mobile/alert_maneuver_request.cc]: @okozlovlux you missed logic from first version. Is is correct?

**Comment by JackLivio** [comment]
@LuxoftAKutsan Does this PR resolve #980

**Pull request merged**